materialize-snowflake/client.go

```diff
@@ -56,6 +56,12 @@ func (c *client) PutSpec(ctx context.Context, updateSpec sql.MetaSpecsUpdate) er
 }
 
 func (c *client) CreateTable(ctx context.Context, tc sql.TableCreate) error {
+	if len(tc.Path) > 1 {
```

Comment:
I think this is OK for a quick-fix since it sounds like there are users that have an immediate need for automatic schema creation.

As a broader feature, I think we will want `InfoSchema` to represent which schemas already exist, and materialization `(*client).InfoSchema` implementations to populate that listing of existing schemas. Then we could add a `CreateSchema` or similar to the `boilerplate.Applier` interface and have the boilerplate figure out if there are schemas that don't exist but need to & create them and call `CreateSchema` if so.

Reply:
@williamhbaker I don't think schemas are a universal-enough of a concept among materialization technologies for us to incorporate this into `InfoSchema`, which is part of `materialize-boilerplate`, not `materialize-sql`.

If this was a concept in `materialize-sql` I think it would make sense, since most of our current materialize-sql connectors do have schemas (I think Firebolt doesn't)